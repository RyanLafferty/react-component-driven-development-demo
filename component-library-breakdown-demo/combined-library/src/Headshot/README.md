### Usage
```js
import { Headshot } from 'combined-library';
import headshot from 'headshot-compressed.png';

...

<Headshot headshot={headshot} />
```
