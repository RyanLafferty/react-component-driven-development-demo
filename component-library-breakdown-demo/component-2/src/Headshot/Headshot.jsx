import React from 'react';
import PropTypes from 'prop-types';
import styles from './Headshot.module.scss';

const Headshot = ({ headshot }) => (
  <img src={headshot} className={styles.container} />
);

Headshot.propTypes = {
  headshot: PropTypes.string,
};

export default Headshot;
