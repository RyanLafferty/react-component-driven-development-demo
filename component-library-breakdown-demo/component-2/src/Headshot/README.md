### Usage
```js
import Headshot from 'component-2';
import headshot from 'headshot-compressed.png';

...

<Headshot headshot={headshot} />
```
