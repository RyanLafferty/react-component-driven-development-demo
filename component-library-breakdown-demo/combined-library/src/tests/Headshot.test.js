import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Headshot from 'Headshot';

describe('Headshot', () => {
  let props;
  const renderer = new ShallowRenderer();

  beforeEach(() => {
    props = {};
  });

  describe('when no props are provided', () => {
    it('Headshot renders', () => {
      expect(
        renderer.render(<Headshot {...props} />),
      ).toMatchSnapshot();
    });
  });
});
