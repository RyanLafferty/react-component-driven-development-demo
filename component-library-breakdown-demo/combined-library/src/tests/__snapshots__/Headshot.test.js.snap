// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headshot when no props are provided Headshot renders 1`] = `
<img
  className="container"
/>
`;
